src/main/java/duke/DukeController.java

```diff
@@ -0,0 +1,140 @@
+
+import java.io.IOException;
+import java.util.Scanner;
+
+public class DukeController {
+
+    private static Task[] tasks = new Task[100];
+    private static int currentTaskLength = 0;
+    private static Scanner sc = new Scanner(System.in);
+    private static String input;
+    private static int currentCommand;
+
+
+    public static void run() throws IOException {
+        DukeIO.initialize();
+        while (true){
+            input = sc.nextLine();
+            currentCommand = DukeCommandValidator.getCommand(input);
+            switch (currentCommand){
+            case DukeCommands.INVALID_COMMAND:{
+                System.out.println("Please enter a valid command!");
+                DukeUI.printMenu();
+                break;
+            }
+            case DukeCommands.LIST:{
+                DukeIO.readDukeData();
+                list();
+                break;
+            }
+            case DukeCommands.EXIT:{
+                return;
+            }
+            case DukeCommands.DONE:{
+                DukeIO.readDukeData();
+                done();
+                DukeIO.writeDukeData();
+                break;
+            }
+            case DukeCommands.ADD:{
+                DukeIO.readDukeData();
+                add();
+                DukeIO.writeDukeData();
+                break;
+            }
+            case DukeCommands.DELETE:{
+                DukeIO.readDukeData();
+                delete();
+                DukeIO.writeDukeData();
+                break;
+            }
+            default: {
+                System.out.println("Unknown error has occurred! Please try again.");
+            }
+            }
+        }
+    }
+
+    public static void list(){
+        for(int i=0; i<currentTaskLength; i++){
+            System.out.printf("%d.", i+1);
+            tasks[i].printTask();
+        }
+    }
+
+    public static void delete(){
+        // ensure task index to delete is in range
+        int taskNumberToDelete = Integer.parseInt(input.substring(7));
+        if(taskNumberToDelete>currentTaskLength || taskNumberToDelete < 1) {
+            System.out.println("The task number is out of range! Please check the " +
+                    "current tasks using the list command and enter a valid task number.");
+            return;
+        }
+        // delete the task
+        Task[] copy = new Task[100];
+        for (int i = 0, j = 0; i < currentTaskLength; i++) {
+            if (i != taskNumberToDelete - 1) {
+                copy[j++] = tasks[i];
+            }else{
+                System.out.println("Noted. I've removed this task:");
+                System.out.println("  ["
+                        + tasks[i].getStatusIcon() + "] "
+                        + tasks[i].getDescription());
+            }
+        }
+        tasks = copy;
+        currentTaskLength--;
+        System.out.printf("Now you have %d tasks in the list.\n", currentTaskLength);
+    }
+
+    public static void done(){
```

Comment:
Consider using more descriptive method names that tells the reader more about what the method actually does; for example, 'markAsDone' instead of just 'done'